### PR TITLE
Implement ground friction for standing entities

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- reduce standing velocities via ground friction in DBSP circuit
- document friction behaviour and mark roadmap entry complete
- test ground friction with rstest and rspec

## Testing
- `make lint`
- `make test`
- `make markdownlint`
- `make nixie` *(fails: FileNotFound: failed copying files from cache to destination for package cytoscape)*

------
https://chatgpt.com/codex/tasks/task_e_689fd370a2b08322a3fffdefba8bd879

## Summary by Sourcery

Apply ground friction to standing entities in the DBSP motion circuit and update tests and documentation to reflect this behaviour.

New Features:
- Scale horizontal velocities by (1 − GROUND_FRICTION) for standing entities in the motion circuit

Enhancements:
- Document the ground friction operator in the physics design docs and mark the roadmap entry as complete
- Ensure all motion logic, including friction, is computed within the DBSP circuit without external duplication

Tests:
- Update existing DBSP and BDD tests to expect reduced velocities due to ground friction
- Add parameterized tests for standing friction with positive, negative, and zero velocities

### CODE REVIEW

File: src/dbsp_circuit/streams.rs [hunk 1]
```diff
@@ -269,9 +269,15 @@ pub(super) fn standing_motion_stream(
     let moved = standing
         .map_index(|pf| (pf.position.entity, pf.position))
         .join(&velocities.map_index(|v| (v.entity, *v)), |_, pos, vel| {
-            let new_x = OrderedFloat(pos.x.into_inner() + vel.vx.into_inner());
-            let new_y = OrderedFloat(pos.y.into_inner() + vel.vy.into_inner());
-            (new_x, new_y, pos.entity, vel.vx, vel.vy)
+            let apply_friction = |v: OrderedFloat<f64>| {
+                // Ground friction opposes horizontal motion without reversing direction.
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/dbsp_circuit/streams.rs]: Has this now been resolved?

File: tests/dbsp_motion_logic.rs [hunk 2]
```diff
@@ -114,3 +114,37 @@ fn motion_cases(
...
+    let vel_out: Vec<NewVelocity> = circuit
+        .new_velocity_out()
+        .consolidate()
+        .iter()
+        .map(|t| t.0)
+        .collect();
+    assert_eq!(vel_out.len(), 1);
+    assert_relative_eq!(vel_out[0].vx.into_inner(), expected_vx);
+    assert_relative_eq!(vel_out[0].vy.into_inner(), 0.0);
+    assert_relative_eq!(vel_out[0].vz.into_inner(), 0.0);
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/dbsp_motion_logic.rs]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- Consider centralizing the ground friction calculation into a shared helper so that both your circuit code and your tests derive velocities from the same function instead of duplicating the formula.
- Add a validation or clamp on GROUND_FRICTION to ensure it remains within [0,1] so you can’t accidentally invert or amplify velocities when the constant changes.
- In your BDD tests, use an approximate float comparison (e.g., assert_relative_eq) for expected velocities rather than exact equality to avoid spurious failures from tiny rounding differences.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?



    59-61: Remove remaining literals tied to friction

    Use apply_ground_friction(1.0) for expected values to avoid breakage if GROUND_FRICTION changes.

    Apply this diff:

    -    Some(Position { entity: 1, x: 0.9.into(), y: 0.0.into(), z: 1.0.into() }),
    -    Some(vel(1, 0.9, 0.0, 0.0)),
    +    Some(Position { entity: 1, x: apply_ground_friction(1.0).into(), y: 0.0.into(), z: 1.0.into() }),
    +    Some(vel(1, apply_ground_friction(1.0), 0.0, 0.0)),

117-150: Assert position translation in standing_friction

Strengthen the test by asserting x is translated by the friction-applied velocity and z snaps to floor.

Apply this diff:

     circuit.step().expect("circuit step failed");
 
+    let pos_out: Vec<NewPosition> = circuit
+        .new_position_out()
+        .consolidate()
+        .iter()
+        .map(|t| t.0)
+        .collect();
+    assert_eq!(pos_out.len(), 1);
+    assert_relative_eq!(pos_out[0].x.into_inner(), apply_ground_friction(vx));
+    assert_relative_eq!(pos_out[0].y.into_inner(), 0.0);
+    assert_relative_eq!(pos_out[0].z.into_inner(), 1.0);
+
     let vel_out: Vec<NewVelocity> = circuit

43-45: Replace hard-coded 0.9 with the helper

Derive expected position and velocity from apply_ground_friction rather than baking the current constant into the test.

Apply this diff:

-    Some(Position { entity: 1, x: 0.9.into(), y: 0.0.into(), z: 1.0.into() }),
-    Some(vel(1, 0.9, 0.0, 0.0)),
+    Some(Position { entity: 1, x: apply_ground_friction(1.0).into(), y: 0.0.into(), z: 1.0.into() }),
+    Some(vel(1, apply_ground_friction(1.0), 0.0, 0.0)),
